storage/storage.go

```diff
@@ -0,0 +1,71 @@
+package storage
+
+import (
+	"strconv"
+	"sync"
+
+	"github.com/artemis13/platform-go-challenge/models"
+)
+
+var (
+	users = make(map[uint]*models.User)
+	mu    sync.RWMutex
+)
```

Comment:
Would there be a benefit to creating a struct to encapsulate these and use that struct as a receiver for the following functions?

Reply:
Yes, by creating a struct to encapsulate the users map and the mu mutex we can make use of better software design and development practices. We ensure better code management and future code expansions are easier to manage. Also we can control better unauthorizes access to the users map. Also we can centralizing the synchronization logic within the struct and avoid race conditions and data corruptions.